hypertrace-trace-enricher-impl/src/test/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/BackendEntityResolverTest.java

```diff
@@ -293,8 +289,21 @@ public void checkBackendEntityGeneratedFromHttpEventUrlWithIllegalCharacter() {
                 .setEventId(ByteBuffer.wrap("random_event_id".getBytes()))
                 .setRefType(EventRefType.CHILD_OF).build())).build();
 
-    Optional<Entity> backendEntity = backendEntityResolver.resolveEntity(e, structuredTraceGraph);
-    assertTrue(backendEntity.isEmpty());
```

Comment:
Parsing the URL field specified in this test case with `URI` throws an exception which is handled internally and returns an empty object. But these could be potential security threats, which is why we should be able to parse them. URL parses such strings, as demonstrated in this updated test case.